### PR TITLE
dogfood our own gcloud image

### PR DESCRIPTION
**Pull request by imjasonh**
This has the usual potential bootstrapping issues, which we already have by dogfooding our CSI secrets-store images, and which has the same solution if it ever becomes a problem -- remove the tag and we'll use the official upstream gcloud image instead.

### CODE REVIEW

**Review by ajayk — DISMISSED** [review]

File: .github/workflows/dag-push-production.yaml
Comment (mattmoor) [inline .github/workflows/dag-push-production.yaml:139]: Can you pin this and let digestabot float it?  This gives us a rollback story when stuff breaks 🙏

**Comment by imjasonh** [comment]
I'd still like to get @kaniini 's signoff on this before we merge please 🙏

**Review by kaniini — APPROVED** [review]

**Pull request merged**
